src/h5web/visualizations/heatmap/utils.ts

```diff
@@ -1,12 +1,18 @@
-import { scaleLinear } from 'd3-scale';
+import { scaleLinear, scaleSymlog } from 'd3-scale';
 import { range } from 'lodash-es';
-import { D3Interpolator } from './models';
+import { D3Interpolator, Domain, DataScale } from './models';
 
 export const adaptedNumTicks = scaleLinear()
   .domain([300, 900])
   .rangeRound([3, 10])
   .clamp(true);
 
+export function getDataScale(domain: Domain, isLog: boolean): DataScale {
+  const scale = (isLog ? scaleSymlog : scaleLinear)();
+  scale.domain(domain);
+  return scale;
+}
```

Comment:
If I understand well, you had to make `useDataScale` a regular function rather than a hook to be able to call it from the worker that is not a hook ?

Reply:
I would have liked to pass the scale returned by `useDataScale` to the web worker, but as I mentioned, you can't pass functions to a web worker. https://github.com/GoogleChromeLabs/comlink/blob/master/structured-clone-table.md